hazelcast/src/main/java/com/hazelcast/json/internal/JsonSchemaHelper.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.json.internal;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.json.JsonReducedValueParser;
+import com.hazelcast.internal.json.JsonValue;
+import com.hazelcast.internal.json.ParseException;
+import com.hazelcast.internal.serialization.impl.NavigableJsonInputAdapter;
+import com.hazelcast.query.impl.getters.JsonPathCursor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+
+/**
+ * Responsible for creating and using {@link JsonSchemaNode}s.
+ */
+public final class JsonSchemaHelper {
+
+    private JsonSchemaHelper() {
+        // private constructor
+    }
+
+    /**
+     * Creates a {@link JsonPattern} for given json value. If given path
```

Comment:
Minor: maybe say that it creates a pattern for the given query path and that if the path does not match the schema or a terminal value in the schema, null is returned.